### PR TITLE
fix: fixed NPE when metadata is empty in a booster.yaml file

### PR DESCRIPTION
**Pull request by gastaldi**
An empty metadata tag causes the following exception:

```
Caused by: java.lang.NullPointerException
  | at java.util.Collections$UnmodifiableMap.<init>(Collections.java:1446)
  | at java.util.Collections.unmodifiableMap(Collections.java:1433)
  | at io.fabric8.launcher.booster.catalog.Booster.getMetadata(Booster.java:260)
  | at io.fabric8.launcher.web.endpoints.BoosterCatalogEndpoint.lambda$getRuntimes$0(BoosterCatalogEndpoint.java:110)
  | at java.util.Optional.ifPresent(Optional.java:159)
  | at io.fabric8.launcher.web.endpoints.BoosterCatalogEndpoint.getRuntimes(BoosterCatalogEndpoint.java:108)
  | at io.fabric8.launcher.web.endpoints.BoosterCatalogEndpoint$Proxy$_$$_WeldSubclass.getRuntimes(Unknown Source)
  | at io.fabric8.launcher.web.endpoints.BoosterCatalogEndpoint$Proxy$_$$_WeldClientProxy.getRuntimes(Unknown Source)
 

```

### CODE REVIEW

File: src/main/java/io/fabric8/launcher/booster/catalog/Booster.java [hunk 1]
```diff
@@ -257,7 +257,10 @@ public Booster forEnvironment(String environmentName) {
      */
     @SuppressWarnings("unchecked")
     public Map<String, Object> getMetadata() {
-        return data.containsKey(KEY_METADATA) ? Collections.unmodifiableMap((Map<String, Object>) data.get(KEY_METADATA)) : Collections.emptyMap();
+        Map<String, Object> metadata = (Map<String, Object>) data.get(KEY_METADATA);
```
Comment (quintesse) [inline src/main/java/io/fabric8/launcher/booster/catalog/Booster.java]: I don't get why this change is necessary. How can `get(x)` return `null` if `containsKey(x)` was `true`? Is someone somewhere doing a `.put(x, null)`? If so that's really what should be fixed I think.

Comment (gastaldi) [reply src/main/java/io/fabric8/launcher/booster/catalog/Booster.java]: The YAML parser automatically puts `null` in the `metadata` key when if finds a `booster.yaml` like this: https://github.com/fabric8-launcher/launcher-booster-catalog/blob/v39/spring-boot/current-redhat/health-check/booster.yaml#L7

Comment (gastaldi) [reply src/main/java/io/fabric8/launcher/booster/catalog/Booster.java]: If you run `BoosterTest`, you will see the NPE happening

Comment (gastaldi) [reply src/main/java/io/fabric8/launcher/booster/catalog/Booster.java]: It seems that `BoosterTest` is conflicting with other tests. I don't exactly know why, but I'm working on it

**Comment by gastaldi** [comment]
This is the error I get while debugging: 

```
May 24, 2018 10:49:51 PM io.fabric8.launcher.booster.catalog.spi.NativeGitBoosterCatalogPathProvider cloneRepository
INFO: Executing: git clone http://localhost:8765/snowdrop/spring-boot-cache-booster --branch 1.5.12-2 --recursive --depth=1 -c advice.detachedHead=false /home/ggastald/workspace/launchpad/launcher-booster-catalog-service/target/booster-catalog88289159988031765/.boosters/spring-boot_current-community_cache_booster
Cloning into '/home/ggastald/workspace/launchpad/launcher-booster-catalog-service/target/booster-catalog88289159988031765/.boosters/spring-boot_current-community_cache_booster'...
fatal: remote error: Git repository not found
```

@quintesse any idea?

**Comment by gastaldi** [comment]
Ah I think I got this, let me confirm it
